src/github.com/mozilla-services/pushgo/simplepush/config_test.go

```diff
@@ -94,9 +94,15 @@ func TestConfigFile(t *testing.T) {
 	}
 	origin, _ := url.ParseRequestURI("https://push.services.mozilla.com")
 	origins := []*url.URL{origin}
-	if !reflect.DeepEqual(origins, app.sh.origins) {
-		t.Errorf("Mismatched origins: got %#v; want %#v",
-			app.sh.origins, origins)
+	socketHandler := app.SocketHandler()
+	if sh, ok := socketHandler.(*SocketHandler); ok {
```

Comment:
I completely understand why and the approach, but, yeah, all the alarm bells are ringing in my head about this too. Unit tests are fun ways to remove type safety in Go.